src/components/cuda/cupti_common.c

```diff
@@ -356,9 +375,39 @@ void cuptic_disabled_reason_get(const char **pmsg)
     *pmsg = cuptic_disabled_reason_g;
 }
 
+static int dl_iterate_phdr_cb(struct dl_phdr_info *info, __attribute__((unused)) size_t size, __attribute__((unused)) void *data)
+{
+    const char *library_name = "libcudart.so";
+    const char *library_path = info->dlpi_name;
+
+    if (library_path != NULL && strstr(library_path, library_name) != NULL) {
+        linked_cudart_path = strdup(dirname(dirname((char *) library_path)));
+    }
+
+    return PAPI_OK;
+}
+
+static int get_user_cudart_path(void)
+{
+    dl_iterate_phdr(dl_iterate_phdr_cb, NULL);
+    if (linked_cudart_path != NULL) {
+        return PAPI_OK;
+    }
+    else {
+        return PAPI_EMISC;
+    }
```

Comment:
This is a bit awkward as it is written. Can you change it to:
```
if (NULL == linked_cudart_path) {
    return PAPI_EMISC;
}
return PAPI_OK;
```

Reply:
Done